Explorer/Assets/Scripts/ECS/Unity/Textures/Components/VideoTextureConsumer.cs

```diff
@@ -17,12 +59,23 @@ public struct VideoTextureConsumer : IDisposable
         public VideoTextureConsumer(Texture2D texture)
         {
             Texture = new Texture2DData(texture);
+            renderers = new List<MeshRenderer>();
         }
 
         public void Dispose()
         {
             // On Dispose video textures are dereferenced by material that acquired it
             Texture = null!;
+            renderers.Clear();
+        }
+
+        /// <summary>
+        /// Stores a reference to a renderer that consumes the same texture.
+        /// </summary>
+        /// <param name="renderer">The renderer using the video texture.</param>
+        public void AddConsumerMeshRenderer(MeshRenderer renderer)
```

Comment:
There should be somewhere removal of consumers as `VideoTextureConsumer` can outlive any given consumers if multiple consumers are connected

Reply:
In the beginning I added a method to remove renderers but then I didn't find the proper place from where to call it.

Reply:
It should be the same moment when `Texture2DData.Dereference`, try to search for it, if you can't find, reach out to me, we will take a look together

Reply:
I saw there is a TryAddConsumer extension method but not "RemoveConsumer". For removing renderers, the only thing I found related to this was the VideoTextureConsumer.Dispose which is only called from the CleanUpMediaPlayerSystem. The Dereference method of StreamableRefCountData<Texture2D> (base of Texture2DData) is called from places that don't know anything about VideoTextureConsumers or even textures, and there is no reference in the Texture2DData to the VideoTextureConsumers that could free resources if the Dereference method was overriden.